pkg/api/handlers/mcp.go

```diff
@@ -213,29 +240,36 @@ func (m *MCPHandler) GetServer(req api.Context) error {
 
 func (m *MCPHandler) DeleteServer(req api.Context) error {
 	var (
-		server v1.MCPServer
-		id     = req.PathValue("mcp_server_id")
+		server    v1.MCPServer
+		id        = req.PathValue("mcp_server_id")
+		catalogID = req.PathValue("catalog_id")
 	)
 
-	project, err := getProjectThread(req)
-	if err != nil {
+	if err := req.Get(&server, id); err != nil {
```

Comment:
Is it okay that this got moved to above where we check for the project thread? Or was it important that the project check happen prior to getting the server from the db?

Reply:
This should be fine. We only need the project to ensure chatbot users don't delete MCP servers in the agent project later on in the method, and getting the mcp server first shouldn't affect that check.